src/py/flwr/superexec/exec_servicer.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""SuperExec API servicer."""
+
+
+from logging import ERROR, INFO
+from subprocess import Popen
+from typing import Dict
+
+import grpc
+
+from flwr.common.logger import log
+from flwr.proto import exec_pb2_grpc  # pylint: disable=E0611
+from flwr.proto.exec_pb2 import (  # pylint: disable=E0611
+    StartRunRequest,
+    StartRunResponse,
+)
+
+from .executor import Executor
+
+
+class ExecServicer(exec_pb2_grpc.ExecServicer):
+    """SuperExec API servicer."""
+
+    def __init__(self, executor: Executor) -> None:
+        self.executor = executor
+        self.runs: Dict[int, Popen] = {}  # type: ignore
```

Comment:
Should this rather hold the `RunTracker`, assuming we add more content to it over time?
```suggestion
        self.runs: Dict[int, RunTracker] = {}  # type: ignore
```

Reply:
@danieljanes Good point, if we use `RunTracker`, should we make it a `list` instead of a `Dict` as `RunTracker` already contains the `run_id`?

Reply:
@charlesbvll fine to keep it a dict to ensure quick lookup when receiving requests that contain a `run_id`